core/src/test/java/com/datastax/oss/driver/internal/core/pool/ChannelPoolReconnectTest.java

```diff
@@ -133,4 +136,59 @@ public void should_reconnect_when_channel_starts_graceful_shutdown() throws Exce
 
     factoryHelper.verifyNoMoreCalls();
   }
+
+  @Test
+  public void should_let_current_attempt_complete_when_reconnecting_now()
```

Comment:
I can't quite recall which test was causing this, but I think it happened on suite tear down.  It happens pretty consistently when our integration tests run, so I'll see if I can pin that down.